Updates to CONTRIBUTOR_ROLES based on recent nominations (#297)
instructlab/community: CONTRIBUTOR_ROLES.md

* [`ilab` command-line interface (CLI) tool](https://github.com/instructlab/instructlab). This repository is responsible for the the `ilab` CLI tool.
* [taxonomy tree](https://github.com/instructlab/taxonomy). This repository is responsible for the taxonomy tree that allows you to create models tuned with your data.

> This document outlines a core number of contributor roles for InstructLab projects, such as _Member_, _Triager_, _Reviewer_, and _Maintainer_. An _Oversight Committee_ also exists to supervise the overall InstructLab project and its health. Using transparent criteria, the journey between roles is based on individual participation.

Reviewer: ```suggestion
This document outlines a core number of contributor roles for InstructLab projects, including _Member_, _Triager_,  and _Maintainer_. An _Oversight Committee_ also exists to supervise the overall InstructLab project and its health. Using transparent criteria, the journey between roles is based on individual participation.
```

  * You provide code that consistently pass tests.
  * You consistently address bugs or issues that are discovered after code has been accepted.

> **Note:** Members who frequently contribute code are expected to proactively perform code reviews and work towards becoming a Reviewer for the project that they are active in.

Reviewer: While you're at it ....

```suggestion
> [!NOTE]
> Members who frequently contribute code are expected to proactively perform code reviews and work towards 
> becoming a Reviewer for the project that they are active in.
```

* You have directly contributed to the project through implementation and/or review.
* You have been sponsored by two Maintainers.

> If you meet the requirements, have one of your sponsors open an issue in the relevant repository with the provided nomination template.

Reviewer: ```suggestion
One of the sponsors should open an issue in the relevant repository to provide a nomination. The issue should use the provided nomination template if one exists in that repo.
```

Author: Appending this text as well: `If one does not exist, the sponsor should create a template prior to creating the nomination.`

> Maintainers will vote publically on the issue, responding with either acceptance or feedback for suggested improvement. Any concerns may be discussed privately amongst the existing Maintainer team.

Reviewer: ```suggestion
Maintainers will vote publically on the issue, expressing their support via a GitHub comment or emoji reaction to the nomination summary. Any concerns may be discussed privately amongst the existing Maintainer team. If feedback needs to be given to the nominee, the sponsor should provide that feedback privately.
```

> Maintainers will vote publically on the issue, expressing their support via a GitHub comment or emoji reaction to the nomination summary. Any concerns may be discussed privately amongst the existing Maintainer team. If feedback needs to be given to the nominee, the sponsor should provide that feedback privately.

Reviewer 2: [publically](https://en.wiktionary.org/wiki/publically) is not a wrong word, but `publicly` is more commonly used, and "is generally considered more correct"

Not requesting a change, more of a note as I was wondering why `publically` wasn't in the dictionary. ("This spelling is omitted from many dictionaries")

Author: Oh that's my bad! I will change it @cybette thanks for the heads up!
